yt/utilities/answer_testing/testing_utilities.py

```diff
@@ -0,0 +1,578 @@
+    # parameters
+    # Use a new dictionary so as to not overwrite the non-hashed test
+    # results in case those are to be saved
+    hashed_results = {}
+    for test_name, test_value in results.items():
+        hashed_results[test_name] = generate_hash(test_value)
+    return hashed_results
+
+
+def _hash_dict(data):
+    r"""
+    Specifically handles hashing a dictionary object.
+
+    Parameters
+    ----------
+    data : dict
+        The dictionary to be hashed.
+
+    Returns
+    -------
+    hd.hexdigest : str
+        The hex digest of the hashed dictionary.
+    """
+    hd = None
+    for key, value in data.items():
+        # Some keys are tuples, not strings
+        if not isinstance(key, str):
+            key = key.__repr__()
+        # Test suites can return values that are dictionaries of other tests
+        if isinstance(value, dict):
+            hashed_data = _hash_dict(value)
+        else:
+            hashed_data = bytearray(key.encode("utf8")) + bytearray(value)
+        # If we're returning from a recusive call (and therefore hashed_data
+        # is a hex digest), we need to encode the string before it can be
+        # hashed
+        if isinstance(hashed_data, str):
+            hashed_data = hashed_data.encode("utf8")
+        if hd is None:
+            hd = hashlib.md5(hashed_data)
+        else:
+            hd.update(hashed_data)
+    return hd.hexdigest()
+
+
+def generate_hash(data):
+    r"""
+    Actually performs the hash operation.
+
+    Parameters
+    ----------
+    data : python object
+        Data to be hashed.
+
+    Returns
+    -------
+    hd : str
+        Hex digest of the hashed data.
+    """
+    # Sometimes md5 complains that the data is not contiguous, so we
+    # make it so here
+    if isinstance(data, np.ndarray):
+        data = np.ascontiguousarray(data)
+    elif isinstance(data, str):
+        data = data.encode("utf-8")
+    # Try to hash. Some tests return hashable types (like ndarrays) and
+    # others don't (such as dictionaries)
+    try:
+        hd = hashlib.md5(data).hexdigest()
+        # Handle those tests that return non-hashable types. This is done
+        # here instead of in the tests themselves to try and reduce boilerplate
+        # and provide a central location where all of this is done in case it needs
+        # to be changed
+    except TypeError:
+        if isinstance(data, dict):
+            hd = _hash_dict(data)
+        elif data is None:
+            hd = hashlib.md5(bytes(str(-1).encode("utf-8"))).hexdigest()
+        else:
+            raise
+    return hd
+
+
+def _save_result(data, outputFile):
+    r"""
+    Saves the test results to the desired answer file.
+
+    Parameters
+    ----------
+    data : dict
+        Test results to be saved.
+
+    outputFile : str
+        Name of file to save results to.
+    """
+    with open(outputFile, "a") as f:
+        yaml.dump(data, f)
+
+
+def _compare_result(new_data, old_data):
```

Comment:
Is that method really necessary? Why not simple `==`?

```
>>> a = dict(a=1,b=2)
>>> b = dict(a=1,b=2)
>>> a == b
True
>>> id(a) == id(b)
False
```

Reply:
I got rid of the method and added the `==` directly to the hashing fixture